### PR TITLE
Add label textbox color

### PR DESCRIPTION
**Pull request by AndreiMoraru123**
## Context

PR for: https://github.com/pytorch/vision/issues/9199#issuecomment-3249895474

## How to use

```py
import torch
from PIL import Image
from torchvision.utils import draw_bounding_boxes

boxes = torch.tensor(
    [[100, 100, 150, 150], [50, 50, 100, 100], [300, 100, 400, 300]],
    dtype=torch.float32,
)

image = torch.ones((3, 500, 500), dtype=torch.uint8) * 255

colors = ["blue", "yellow", "red"]
label_bg_colors = ["green", "magenta", "orange"]
image = draw_bounding_boxes(
    image,
    boxes,
    colors=colors,
    width=1,
    fill=False,
    labels=colors,
    label_background_colors=label_bg_colors,  # new param
    fill_labels=True,  # only does something when fill_labels is set to True
)
img = Image.fromarray(image.permute(1, 2, 0).numpy())
img.show()
```

image

### CODE REVIEW

File: torchvision/utils.py [hunk 1]
```diff
@@ -362,6 +364,11 @@ def draw_bounding_boxes(
     else:
         label_colors = colors.copy()  # type: ignore[assignment]
 
+    if fill_labels:
+        label_background_colors = _parse_colors(label_background_colors, num_objects=num_boxes) if label_background_colors else colors.copy()  # type: ignore[assignment]
+    else:
+        label_background_colors = colors.copy()  # type: ignore[assignment]
```
Comment (AntoineSimoulin) [inline torchvision/utils.py]: Maybe we can add the conditions ` if label_background_colors` in the first `if fill_labels` for more clarity here?

File: torchvision/utils.py [hunk 2]
```diff
@@ -320,6 +320,8 @@ def draw_bounding_boxes(
         font_size (int): The requested font size in points.
         label_colors (color or list of colors, optional): Colors for the label text.  See the description of the
             `colors` argument for details.  Defaults to the same colors used for the boxes, or to black if ``fill_labels`` is True.
+        label_background_colors (color or list of colors, optional): Colors for the label text box fill. Defaults to the
+            same colors used for the boxes. Ignored when ``fill_labels`` is False.
```
Comment (AntoineSimoulin) [inline torchvision/utils.py:324]: Do we really ignore the `label_background_colors` when ``fill_labels`` is False?

Comment (AndreiMoraru123) [reply torchvision/utils.py:324]: Do we not want to? We could choose to draw the outline only, but the code is currently only drawing the fill.

 If the existing code at line 406: https://github.com/pytorch/vision/pull/9204/files#diff-1aec34e36ed0b4f955a463c517f7a8d5df70f8fa8118594672471e8c2b6d8b17R406 were like this

```py
  draw.rectangle(
      (left - box_margin, top - box_margin, right + box_margin, bottom + box_margin), fill=..., outline=...
  )
```

Then we could still draw the outline. Though that would mean changing a bit the previous behaviour (maybe users wouldn't mind about something as small as this). What do you think?

File: torchvision/utils.py (hunk 1 above)
Comment (AndreiMoraru123) [reply torchvision/utils.py]: I'd like to. I could think of two ways, and I like neither😅

```py
if fill_labels:
    label_colors = _parse_colors(label_colors if label_colors else "black", num_objects=num_boxes)
    label_background_colors = _parse_colors(label_background_colors if label_background_colors else colors.copy(), num_objects=num_boxes)
else:
    label_colors = _parse_colors(label_colors if label_colors else "black", num_objects=num_boxes) if label_colors else colors.copy()
    label_background_colors = colors.copy()
```

or

```py
if fill_labels:
    label_colors = _parse_colors(label_colors if label_colors else "black", num_objects=num_boxes)  # type: ignore[assignment]
    label_background_colors = _parse_colors(label_background_colors, num_objects=num_boxes) if label_background_colors else colors.copy()  # type: ignore[assignment]
else:
    if label_colors:
        label_colors = _parse_colors(label_colors if label_colors else "black", num_objects=num_boxes)  # type: ignore[assignment]
    else:
        label_colors = colors.copy()
    label_background_colors = colors.copy() 
```

because of the level of nesting they introduce. Maybe it would be best to rethink the original branching logic altogether?

**Comment by AntoineSimoulin** [comment]
Also, is it possible to apply linting with `pre-commit run --all-files` as detailed in the [contributing guidelines](https://github.com/pytorch/vision?tab=contributing-ov-file#pre-commit-hooks)?

**Comment by AndreiMoraru123** [comment]
> Also, is it possible to apply linting with `pre-commit run --all-files` as detailed in the [contributing guidelines](https://github.com/pytorch/vision?tab=contributing-ov-file&rgh-link-date=2025-09-04T17%3A52%3A55Z#pre-commit-hooks)?

Must've missed the memo. I noticed now the repo has flake8 and ufmt. Hopefully, running the pre-commit on the modified files should be enough in this case.

File: torchvision/utils.py [hunk 3]
```diff
@@ -320,6 +320,8 @@ def draw_bounding_boxes(
...
         fill_labels (bool): If `True` fills the label background with specified box color (from the ``colors`` parameter). Default: False.
```
Comment (NicolasHug) [inline torchvision/utils.py]: I think the description of this parameter needs to be updated :)

File: torchvision/utils.py (hunk 1 above)
Comment (AntoineSimoulin) [reply torchvision/utils.py]: Oh yeah seems to complex to group the two. I just meant, if we do something like that (just add the check if label_background_colors is not None in the first if).
```python
if fill_labels and label_background_colors:
    label_background_colors = _parse_colors(label_background_colors, num_objects=num_boxes)  # type: ignore[assignment]
else:
    label_background_colors = colors.copy()  # type: ignore[assignment]
```

File: torchvision/utils.py (hunk 2 above)
Comment (AntoineSimoulin) [reply torchvision/utils.py:324]: No I think you were right initially, let's keep it like that!

File: torchvision/utils.py (hunk 1 above)
Comment (AndreiMoraru123) [reply torchvision/utils.py]: Yeah, this looks much more straightforward, adjusted it in [`19ae698` (#9204)](https://github.com/pytorch/vision/pull/9204/commits/19ae698d79bc6527fb82b8ab0d5e12a10e4fb16f)

File: torchvision/utils.py (hunk 3 above)
Comment (AntoineSimoulin) [reply torchvision/utils.py]: @NicolasHug are you aligned with the current state? Can we resolve this comment?
